Update READMEs to help folks figure out how to successfully make docs (#29865)
hashicorp/terraform: website/README.md

**Set Up Local Environment**

1. [Install Docker](https://docs.docker.com/get-docker/).
2. Create a `~/go` directory manually or by [installing Go](https://golang.org/doc/install).
3. Open terminal and set `GOPATH` as an environment variable:

   Bash: `export $GOPATH=~/go`(bash)

   Zsh: `echo -n 'export GOPATH=~/go' >> ~/.zshrc`

4. Restart your terminal or command line session.

**Launch Site Locally**

1. Navigate into your local `terraform` top-level directory and run `make website`.
1. Open `http://localhost:3000` in your web browser. While the preview is running, you can edit pages and Next.js will automatically rebuild them.
1. When you're done with the preview, press `ctrl-C` in your terminal to stop the server.

> cccccckebtefdtliheunihirifuucdchterukkjehlrr

Author: ```suggestion
```

### Deployment

> New commits in `hashicorp/terraform` do not automatically deploy the site. To use the latest upstream content, you'll need to open a PR bumping the submodule commit. If your changes aren't being deployed, it's very likely that you need to open a PR to update the submodule commit.

Author: @thiskevinwang - this is where I could use your help!

### Deployment

New commits in `hashicorp/terraform` do not automatically deploy the site. Do the following for documentation pull requests:

> - **Add a backport label to the PR.** This is the label corresponding to the latest Terraform patch release (e.g., `1.2-backport`). When you merge your PR to `main`, GitHub bot automatically generates a backport PR to merge your commits into the appropriate release branch.

Author: ```suggestion
- **Add a backport label to the PR.** Use the label that corresponds to the latest Terraform patch release (e.g., `1.2-backport`). When you merge your PR to `main`, GitHub bot automatically generates a backport PR to merge your commits into the appropriate release branch.
```

> - **Merge the backport PR.** When all tests pass successfully, merge the backport PR into the release branch. The new content will be added to the site during the next minor release.

Reviewer: Simply noting / no action needed:

Once we begin displaying multiple versions for `terraform`, merging backport PRs for **older** versions will update their associated versions on the site within the hour.

Backport PRs for the current, latest version will not affect the latest version on the website. This is so that docs changes may be _queued up_ for the next patch release.

> - **Cherry-pick changes to `stable-website`.** If you want your changes to show up immediately, check out the latest version of the`stable-website` branch, cherry-pick your changes, and run `git push` to add your changes to the remote `stable-website` branch. Your changes will be live on the site within the hour. 

Author: ```suggestion
- **Cherry-pick changes to `stable-website`.** If you want your changes to show up immediately, check out the latest version of the`stable-website` branch, cherry-pick your changes to the branch, and run `git push`. Your changes will be live on the site within the hour. 
```
